ibis/backends/snowflake/__init__.py

```diff
@@ -238,3 +242,13 @@ def list_databases(self, like=None) -> list[str]:
                 "SELECT database_name FROM information_schema.databases"
             ).scalars()
         return self._filter_with_like(databases, like)
+    
+    def _register_in_memory_table(self, _: ops.InMemoryTable) -> None:
+        with self.begin() as con:
+            write_pandas(conn=con.connection.connection,
```

Comment:
```suggestion
            write_pandas(conn=con.connection,
```

You can use the SQLAlchemy proxy object here.

Reply:
sadly I cannot do that without modifying the underlying snowflake-sqlalchemy.

even snowflake snowflake-connector-python seem use this ugly hack :-/

https://github.com/snowflakedb/snowflake-connector-python/blob/21ddcbe87624ae99e71597041fee6fe7d27f4b2f/src/snowflake/connector/pandas_tools.py#L373

Reply:
Oh, interesting, ok then!